homeassistant/components/alexa/smart_home.py

```diff
@@ -1276,6 +1356,158 @@ def async_api_previous(hass, config, request, entity):
     return api_message(request)
 
 
+def api_error_temp_range(request, temp, min_temp, max_temp, unit):
+    """Create temperature value out of range API error response.
+
+    Async friendly.
+    """
+    temp_range = {
+        'minimumValue': {
+            'value': min_temp,
+            'scale': API_TEMP_UNITS[unit],
+        },
+        'maximumValue': {
+            'value': max_temp,
+            'scale': API_TEMP_UNITS[unit],
+        },
+    }
+
+    msg = 'The requested temperature {} is out of range'.format(temp)
+    _LOGGER.info(msg)
+    return api_error(
+        request,
+        error_type='TEMPERATURE_VALUE_OUT_OF_RANGE',
+        error_message=msg,
+        payload={'validRange': temp_range},
+    )
+
+
+def temperature_from_object(temp_obj, to_unit, absolute=True):
+    """Get temperature from Temperature object in requested unit."""
+    from_unit = TEMP_CELSIUS
+    temp = float(temp_obj['value'])
+
+    if temp_obj['scale'] == 'FAHRENHEIT':
+        from_unit = TEMP_FAHRENHEIT
+    elif temp_obj['scale'] == 'KELVIN':
+        # convert to Celsius if absolute temperature
+        if absolute:
+            temp -= 273.15
+
+    if not absolute:
+        if from_unit != to_unit:
+            if from_unit == TEMP_FAHRENHEIT:
+                return temp / 1.8
```

Comment:
Please use our temperature utils in `homeassistant.util.temperature`

Reply:
I already use `homeassistant.util.temperature.convert` (as `convert_temperature`) for conversions at the bottom of that function. For `AdjustTemperature` calls I can't pass the `targetSetpointDelta` value directly to `convert` since it will be adjusted for the C/F zero points, so I hardcodeed the scale conversion here. The alternative is to convert the current temperature to the same unit as the delta, add the delta, and convert the result back, which is more expensive (and may cost some precision).

Reply:
okay, let's keep it.

Reply:
I pushed another commit that adds an `interval` boolean parameter to `homeassistant.util.temperature.convert` which causes it to treat the value as a temperature interval, since I suspect this will not be the only consumer to benefit from this. Is that okay?